### DISCUSSION TITLE
Metainfo in CAB files for multiple devices

### DISCUSSION THREAD

**Discussion by DylanVanAssche** [fwupd/fwupd#5079] [Q&A] [answered]
I have sometimes firmware for multiple devices, they have different GUIDs, but use the same firmware.
I tried to put multiple entries here:

```
<provides>
    <!-- DEVICE A -->
    <firmware type="flashed">84f40464-9272-4ef7-9399-cd95f12da696</firmware>
     <!-- DEVICE B -->
   <firmware type="flashed">84f40464-9272-4ef7-9399-cd95f12da697</firmware>
  </provides>
```

However, it seems that fwupd checks both GUIDs as an *AND*, I need an *OR* case.
Is there a way to achieve that or do you need to upload 2 CAB files with each different metainfo files?

**Comment by hughsie** [comment]
I'm a bit confused; `provides` should already be `OR` i.e. fwupd will check if any of the devices have any of the guids in the metadata.

**Reply by DylanVanAssche** [reply]
I was confused between `requires` and `provides`.

- `provides`: default as OR
- `requires`: default as AND unless with HWids separated by `|`

Luckily, the docs are mentioning this: https://lvfs.readthedocs.io/en/latest/metainfo.html#using-chid

**Comment by DylanVanAssche** [comment, marked as answer]
I was confused between `requires` and `provides`.

- `provides`: default as OR
- `requires`: default as AND unless with HWids separated by `|`

Luckily, the docs are mentioning this: https://lvfs.readthedocs.io/en/latest/metainfo.html#using-chid
